Add 'gdal raster polygonize' (port of gdal_polygonize.py) (#12122)
OSGeo/gdal: doc/source/programs/gdal_raster_polygonize.rst

.. _gdal_raster_polygonize_subcommand:

================================================================================
"gdal raster polygonize" sub-command
================================================================================

.. versionadded:: 3.11

.. only:: html

>     Create a polygon feature dataset from a raster dataset

Reviewer: Perhaps "..from a raster band".

Author: fixed

.. Index:: gdal raster polygonize

Synopsis
--------

.. program-output:: gdal raster polygonize --help-doc

Description
-----------

:program:`gdal raster polygonize` creates vector polygons for all connected
regions of pixels in the raster sharing a common pixel value.

> Each polygon is created with an attribute indicating the pixel value of that polygon. The mask
> (either explicit, or implicit through nodata value or alpha band) associated
> to the selected band will be used to determine the validity of pixels.

Reviewer 2: ```suggestion
Each polygon is created with an attribute indicating the pixel value of that polygon. A mask
(either explicit, or implicit through nodata value or alpha band) associated
to the selected band can be used to determine the which pixels should be included in the processing.
```

Author: fixed

The utility can create the output vector datasource if it does not already exist,
otherwise it may append to an existing one.

The utility is based on the ::cpp:func:`GDALPolygonize` function which has additional
details on the algorithm.

The following options are available:

Standard options
++++++++++++++++

.. include:: gdal_options/of_vector.rst

.. include:: gdal_options/co.rst

.. include:: options/lco.rst

.. include:: gdal_options/overwrite.rst

.. option:: --update

    Whether the output dataset must be opened in update mode. Implies that
    it already exists.

.. option:: --overwrite-layer

    Whether overwriting the existing output vector layer is allowed.

.. option:: --append

    Whether appending features to the existing output vector layer is allowed

.. option:: -b, --band <BAND>

>     Picks a particular band to get the DEM from. Defaults to band 1.

Reviewer: Picks a particular band to polygonize. Defaults to band 1.

Author: fixed

.. option:: -l, --nln, --layer <LAYER>

>     Provides a name for the output vector layer. Defaults to "polygonize".

Reviewer 2: Consider exposing `gdal_polygonize` options `-mask` and `-nomask` (as -`-ignore-mask`)

Reviewer 2: I could do this in a follow-up

Author: > Consider exposing `gdal_polygonize` options `-mask` and `-nomask` (as -`-ignore-mask`)

I deliberately didn't expose them as they seemed a bit too specialized to me, at least at that stage. There are many other algorithms where the user can't provide an "external" mask and I'm not sure why polygonize would be an exception

.. option:: --attribute-name <ELEVATION-NAME>

    The name of the field to create (defaults to "DN").

.. option:: --connectedness 4|8

>     Whether to use 4-connectedness or 8-connectedness. Default is 4.

Reviewer 2: ```suggestion
    Whether to use 4-connectedness or 8-connectedness. With 4-connectedness, pixels will be included in the same output feature if they share a side. With 8-connectedness, pixels will also be included in the same output feature if they share a corner. Default is 4.
```

Author: fixed

Advanced options
++++++++++++++++

.. include:: gdal_options/oo.rst

.. include:: gdal_options/if.rst

Examples
--------

.. example::

>    :title: Create a shapefile with polygons for the connected regions of the input GeoTIFF.

Reviewer: ...of band 1 of the input GeoTIFF.
